bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/handler/CloudBridgeHandler.java

```diff
@@ -0,0 +1,312 @@
+
+        ScheduledExecutorService scheduledPool = ThreadPoolManager.getScheduledPool(SalusBindingConstants.BINDING_ID);
+        scheduledPool.schedule(() -> tryConnectToCloud(localSalusApi), 1, MICROSECONDS);
+
+        this.devicePropertiesCache = Caffeine.newBuilder().maximumSize(10_000)
+                .expireAfterWrite(Duration.ofSeconds(config.getPropertiesRefreshInterval()))
+                .refreshAfterWrite(Duration.ofSeconds(config.getPropertiesRefreshInterval()))
+                .build(this::loadPropertiesForDevice);
+        this.scheduledFuture = scheduledPool.scheduleWithFixedDelay(this::refreshCloudDevices,
+                config.getRefreshInterval() * 2, config.getRefreshInterval(), SECONDS);
+
+        // Do NOT set state to online to prevent it to flip from online to offline
+        // check *tryConnectToCloud(SalusApi)*
+    }
+
+    private void tryConnectToCloud(SalusApi localSalusApi) {
+        try {
+            ApiResponse<SortedSet<Device>> response = localSalusApi.findDevices();
+            if (response.failed()) {
+                @Nullable
+                Error error = response.error();
+                if (error == null) {
+                    error = new Error(500, "Unknown error");
+                }
+                updateStatus(OFFLINE, COMMUNICATION_ERROR, error.code() + ": " + error.message());
+            } else {
+                // there is a connection with the cloud
+                updateStatus(ONLINE);
+            }
+        } catch (Exception ex) {
+            updateStatus(OFFLINE, COMMUNICATION_ERROR,
+                    "@text/cloud-bridge-handler.initialize.cannot-connect-to-cloud [\"" + ex.getMessage() + "\"]");
+        }
+    }
+
+    private void refreshCloudDevices() {
+        logger.debug("Refreshing devices from CloudBridgeHandler");
+
+        try {
+            if (!(thing instanceof Bridge bridge)) {
+                logger.debug("No bridge, refresh cancelled");
+                return;
+            }
+            List<Thing> things = bridge.getThings();
+            for (Thing thing : things) {
+                try {
+                    if (!thing.isEnabled()) {
+                        logger.debug("Thing {} is disabled, refresh cancelled", thing.getUID());
+                        continue;
+                    }
+
+                    @Nullable
+                    ThingHandler handler = thing.getHandler();
+                    if (handler == null) {
+                        logger.debug("No handler for thing {} refresh cancelled", thing.getUID());
+                        continue;
+                    }
+                    thing.getChannels().forEach(channel -> handler.handleCommand(channel.getUID(), REFRESH));
+                } catch (Exception ex) {
+                    logger.warn("Cannot refresh thing {} from CloudBridgeHandler", thing.getUID(), ex);
```

Comment:
What cases would lead to this Exception? I have three comments about this:
1. If custom Exceptions would be used, it could be routed to ThingStatus easily.
2. If the refresh fails, the  ThingStatus should be set to offline, preferably with more details of the why it is offline
3. In line 159 the thing is reported online, while it is not.

Reply:
at 3 - the bridge is reported as online and the exception is only for device. 

I will make the sub device goes offline

Reply:
> What cases would lead to this Exception?

I don't know - that's the point. Similar to: https://github.com/openhab/openhab-addons/pull/16065/files#r1594523925